### PR TITLE
CEO-299 CEO-355 Better method for calculating gridded and random points

### PR DESCRIPTION
**Pull request by sirmspencer**
## Purpose
There were a few errors generating random plots, and one with the samples.  
1. Some plots were out of bounds
2. The spacing was based on size of boundary so they were not generated close enough for larger projects.
3. Random plots could generated in the same location

Simplified some calcs with better reusability. 

## Submission Checklist
- [x] Included Jira issue in the PR title (e.g. `CEO-### <title>`)
- [x] Code passes linter rules (`npm run eslint`/`clj-kondo --lint src`)

## Testing
Basically every test for gridded/random samples/plots.

### CODE REVIEW

**Comment by Oliver-BE** [comment]
Where can I find the tests you mention above? Also, there is a merge conflict.

**Comment by sirmspencer** [comment]
> Where can I find the tests you mention above? Also, there is a merge conflict.

Maybe in the bdd scripts.  create project with random plots, gridded plots, random samples, girdded samples, and center sample.

**Comment by Oliver-BE** [comment]
I am unable to create a project. The website hangs on the "Creating Project" loading screen. I have run the commands to recreate the data base and all the functions. This error occurred multiple times making a standard project with 100 plots and 10 samples per plot (all random distribution).

**Comment by Oliver-BE** [comment]
Random plots work well for me and load quickly now. I'm running into an error with gridded plots, however. It seems like the text that estimates the number of plots is not consistent with the actual number of plots that are generated. Steps to replicate:

### Plot Design
Spatial distribution: Gridded
Plot spacing: 10,000
Plot width: 1,000
Boundary coordinates: 
* North: 43.1432468476
* East: -108.2248982631
* South: 41.5928931176
* West: -110.6477789088
"This project will contain around 648 plots."

### Sample Design
Spatial distribution: Random
Number of samples: 10
"Each plot will contain around 10 samples. There will be around 6,480 total samples in the project."

Creating the project yields: 
"- This action will creation 15795.0 plots. The maximum allowed for the selected plot distribution is 5000.0"

I'm not sure if the text within the Plot Design is wrong here or if the number of plots when going to review the project is calculated incorrectly.
